### ISSUE TITLE
Requesting a new release

### ISSUE THREAD

**Issue by martis42** [bazel-contrib/bazel-mypy-integration#122]
How to integrate this tool into other projects changed much:
- Compatible to modern rules_python
- Users inject mypy
- Basic support for bzlmod

It would be great to depend on all those updates through an official release.

**Comment by jdimatteo** [comment]
Hi I agree a release would help, and in the mean time can someone please point me to some documentation / example with latest / recent rules_python release?

I'm trying to follow the instructions in the WORKSPACE snippet at https://github.com/bazel-contrib/bazel-mypy-integration/releases/tag/0.4.0, and I'm getting the error `'@rules_python//python:pip.bzl' does not contain symbol 'pip_install'`. I think this is caused by [0.4.0 using pip_install here](https://github.com/bazel-contrib/bazel-mypy-integration/blob/0.4.0/repositories/py_repositories.bzl#L6), but [pip_install was dropped from rules_python earlier this year](https://github.com/bazelbuild/rules_python/blob/main/CHANGELOG.md#0280---2024-01-07).

**Comment by adzenith** [comment]
@jdimatteo I would use `main` if at all possible, rather than the 0.4.0 release. As @martis42 pointed out, the last release is very old and these issues have since been fixed.

**Comment by jdimatteo** [comment]
@adzenith yes, I would like to use main, but I was using the 0.4.0 WORKSPACE snippet at https://github.com/bazel-contrib/bazel-mypy-integration/releases/tag/0.4.0. Is there an updated WORKSPACE snippet for latest main commit?

When I try to use the 0.4.0 snippet I get an:

> ERROR: Error computing the main repository mapping: cannot load '@mypy_integration//repositories:deps.bzl': no such file

Can you please point me to updated documentation similar to the 0.4.0's WORKSPACE snippet? Or an example project's WORKSPACE that uses bazel-mypy-integration with rules_python? I don't see any documentation of what to include in my WORKSPACE at https://github.com/bazel-contrib/bazel-mypy-integration/blob/main/README.md.

Any pointers would be greatly appreciated!

**Comment by martis42** [comment]
@jdimatteo the examples https://github.com/bazel-contrib/bazel-mypy-integration/tree/main/examples can be used as blueprint.

This is the snippet how to depend in mypy_integration and initialize its deps:
https://github.com/bazel-contrib/bazel-mypy-integration/blob/main/examples/WORKSPACE#L9-L16

Here the mypy Python dependency is registered:
https://github.com/bazel-contrib/bazel-mypy-integration/blob/main/examples/WORKSPACE#L52-L60

Here mypy is configured (aka providing a config file and the mypy lib):
https://github.com/bazel-contrib/bazel-mypy-integration/blob/main/examples/.bazelrc#L1-L2

**Comment by jdimatteo** [comment]
@martis42 thanks that worked! Didn't mean to hijack this issue requesting a new release, but hopefully these comments help others who might be looking for a release and want to get working on latest main in the mean time.

**Comment by martis42** [comment]
@alexeagle @adzenith 

Do you think we can proceed on this?

One cannot add something to the Bazel central registry without a release artifact existing which can be tested. My proposal for the next steps is:
1. [ ] Review of https://github.com/bazel-contrib/bazel-mypy-integration/pull/131, which does some last cleanups in preparation for `0.5.0`
2. [ ] Creating a pre-release
3. [ ] Add version `0.5.0` of this project to https://github.com/bazelbuild/bazel-central-registry
4. [ ] Make the pre-release a proper release

If a new release is a capacity problem I would volunteer to take over step 3. This is something I am able to do, even as non maintainer.

**Comment by martis42** [comment]
@adzenith Thanks for reviewing my PR. Will it be possible for you to create a pre-release for `0.0.5`?
As offered, I would volunteer to perform the BCR integration if this helps you.

**Comment by JeroenSchmidt** [comment]
Hi @martis42 @adzenith 
As a user of this tool, thank you for what you both are doing for this tool!
